### DISCUSSION TITLE
Defining a base directory for links / composed links

### DISCUSSION THREAD

**Discussion by Eryx5502** [nvim-neorg/neorg#1119] [Q&A]
Hi everyone!

I'm sorry if this has been already answered somewhere, but I searched both the discussions and the issues, as well as the specification and I still haven't found an answer. Let me add I just started using Neorg before asking my question.

Now, suppose that I want to link, from a `.norg` file, have a list of comments on some source code files from a project I'm working on. Let's say that such a project is on `/home/user/code/myproject`, and I want to do a list as in

```norg
~ [First file]{/ /home/user/code/myproject/some-route/first-file}
~ [Second file]{/ /home/user/code/myproject/second-route/second-file}
...
```

Is there a way to define the common part of the link to avoid repeating `/home/user/code/myproject` for each of the links?

Looking for an answer I only found the 'extendable links' on the specs, but couldn't manage to make sense out of them.

**Comment by max397574** [comment]
🤔 I wouldn't know about anything
idk if definitons or variables (advanced syntax element which aren't fully implemented in neorg yet) would be able to do that (@vhyrro ?)

Two things that go in that direction are 1. simply relative links and 2. workspace links

**Reply by Eryx5502** [reply]
If I understand them correctly, relative links could help with a different setup from the one I have right now. I guess it requires to have the `.norg` file inside the project folder, and right now I have the notes in a `~/notes/myproject` folder, configured as a workspace.

**Reply by max397574** [reply]
yes relative links would only be useful if you have the notes in the same folder. Workspace links would be useful when they are in the same workspace
